hash.c

```diff
@@ -6881,93 +6881,161 @@ static const rb_data_type_t env_data_type = {
  *
  *    reviews.length #=> 1
  *
- *  === Default Values
+ *  === Key Not Found?
  *
- *  The methods #[], #values_at and #dig need to return the value associated to a certain key.
- *  When that key is not found, that value will be determined by its default proc (if any)
- *  or else its default (initially `nil`).
+ *  When a method tries to retrieve and return the value for a key and that key <i>is found</i>,
+ *  the returned value is the value associated with the key.
  *
- *  You can retrieve the default value with method #default:
+ *  But what if the key <i>is not found</i>?
+ *  In that case, you have a wide choice of behaviors.
```

Comment:
Is there a wide choice of behaviors? There seems to only be two: default value or raise KeyError.

Reply:
@BurdetteLamar did you miss this comment?

Reply:
Or anything else I may define in a block?

Reply:
But there's really only KeyError or default. AFAIK `nil` is just a special case of the default where the default is nil. Is there a method that returns nil even if `Hash#default` is not?

Reply:
What do you want this to say?

Reply:
Maybe something like?

```suggestion
In that case, certain methods will return a default value while other will raise a \KeyError.
```